rllib/core/learner/learner_group.py

```diff
@@ -176,7 +181,18 @@ def update(
             block: Whether to block until the update is complete.
 
         Returns:
-            A list of dictionaries of results from the updates from the Learner(s)
+            if block is true and reduce_fn is None:
```

Comment:
We need the API description to be higher-level and simpler to understand. It's explaining the code right now.

Reply:
we should just break this into 2 functions instead:

update, and async_update. Multiple return types is weird.

Reply:
Ok let’s do that instead.

Reply:
Let's add a TODO then also for the `reduce_fn`. We have different return types here, too, because of this.
Result reduction should happen on the training_step side.